src/Microsoft.AspNetCore.Razor.Tools/Internal/ResolveTagHelpersDispatchCommand.cs

```diff
@@ -130,36 +115,42 @@ protected override int OnExecute()
             }
         }
 
-        private bool TryResolveFramework(
-            IEnumerable<NuGetFramework> availableFrameworks,
-            out NuGetFramework resolvedFramework)
+        private bool TryResolveFramework(string projectDirectory, out NuGetFramework resolvedFramework)
         {
-            NuGetFramework framework;
+            string frameworkValue;
             if (FrameworkOption.HasValue())
             {
-                var frameworkOptionValue = FrameworkOption.Value();
-                framework = NuGetFramework.Parse(frameworkOptionValue);
+                frameworkValue = FrameworkOption.Value();
+            }
+            else
+            {
+                const string ValueDelimiter = "______FRAMEWORK_______";
+                const string ResolveFrameworkTargetName = "ResolveRazorTargetFramework";
 
-                if (!availableFrameworks.Contains(framework))
-                {
-                    ReportError(
-                        string.Format(
-                            CultureInfo.CurrentCulture,
-                            ToolResources.ProjectDoesNotSupportProvidedFramework,
-                            ProjectArgument.Value,
-                            frameworkOptionValue));
+                var args = new[] { $"/t:{ResolveFrameworkTargetName}" };
+                var command = Command.CreateDotNet("msbuild", args);
+                var outputWriter = new StringWriter();
+
+                var exitCode = command
+                    .ForwardStdErr(Console.Error)
```

Comment:
CaptureStdErr/Out would make more sense here. Invoking `dotnet msbuild` is an internal detail of how the tool gathers info on project context, but and showing MSBuild console output might be misleading to users.